src/liballoc/collections/linked_list.rs

```diff
@@ -878,6 +878,58 @@ impl<T> LinkedList<T> {
         unsafe { self.split_off_after_node(split_node, at) }
     }
 
+    /// Removes the element at the given index and returns it.
+    ///
+    /// This operation should compute in O(n) time.
+    ///
+    /// # Panics
+    /// Panics if at >= len
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(linked_list_remove)]
+    /// use std::collections::LinkedList;
+    ///
+    /// let mut d = LinkedList::new();
+    ///
+    /// d.push_front(1);
+    /// d.push_front(2);
+    /// d.push_front(3);
+    ///
+    /// assert_eq!(d.remove(1), 2);
+    /// assert_eq!(d.remove(0), 3);
+    /// assert_eq!(d.remove(0), 1);
+    /// ```
+    #[unstable(feature = "linked_list_remove", issue = "69210")]
+    pub fn remove(&mut self, at: usize) -> T {
+        let len = self.len();
+        assert!(at < len, "Cannot remove at a nonexistent index");
+
+        if at == 0 {
+            return self.pop_front().unwrap();
+        } else if at == len - 1 {
+            return self.pop_back().unwrap();
+        }
+
+        // Below, we iterate towards the ith node, either from the start or the end,
```

Comment:
```suggestion
        // Below, we iterate towards the node at the given index, either from the start or the end,
```
Or something. Just because we don't have any `i`s anywhere.